text/0009-ed25519-signature-scheme/0009-ed25519-signature-scheme.md

```diff
@@ -0,0 +1,112 @@
+ }
+]
+```
+- bundle hash (81-tryte): RXDECZI9MWEVGUWYELFSPASJQPRKFDEY9ZYKIXFYUBKZYIKOHNMHZJKVBLLNJSRFIRQRIHQZBBVVZRXUX
+- bundle hash (48-byte): bc4c9241451b90473bfd9cb00ddeada30392e43f280a4aa72dee27ab54e9994b297fed74963a2cab33869c1a031af68d
+- public key (32-byte): d75a980182b10ab7d54bfed3c964073a0ee172f3daa62325af021a68f707511a
+- signature (64-byte): 515acc3127aa7bb1075981048c4cbd2fcef8bcefee3e3b0b0d586bddc99da98ef0bf87480eaed229388a5ddb543b9b8c49a61bc7d85668ded4dd94d03309a204
+- signatureMessageFragment (K<sub>tri</sub> || S<sub>tri</sub> 192-tryte): MYICDWA9IVBXJ9HXKYUCY9IYZYSDG9DBNAWZFDN9PZRXHAJA9XB9ZAWDR9G99CZA9CICBYVBLAVXOEBXG9HCHVD9SWVCNYTBDYS9MXJZIZHBEBK9M9GCZDSZZYIWUXUWKZPYNWRCNAZXHYNBBBQWLCQZCCEBGWSWSCRX9AXYNZECWDTZJYSZ9WFYXBI9NXD
+
+# Drawbacks
+
+While the Ed25519 signature algorithm offers a very fast and robust signature scheme, it suffers, like any traditional ECDSA or EdDSA signatures, from susceptibility to Shor's polynomial-time quantum computer algorithm. Although this attack vector is addressed by only publishing the EdDSA public key when signing, it is still inherently less secure than hash-based signature schemes:
+- During the time window, when the signature has been issued to the network but before the transaction has been confirmed and sufficiently finalized, Shor's algorithm could be used to break the private key from the published public key.
+- After the first time an Ed25519 address has been spent from, the public key is also known. As such, this signature scheme offers no stronger quantum robustness for multiple spends from the same address. This is the case, when addresses are either spent from more than once or when the same address is also used in a different fork or testnet of the network.
+
+The signature scheme as a drop-in replacement for the current W-OTS does not offer any replay protection in the current account model: After a transaction has moved funds from A to B, an attacker can simply re-attach this transaction again to transfer the same amount of funds to B a second time, providing a sufficient balance on address A. Although exactly the same attack vector is also present in the current protocol, its effects are considerably worsened as the Ed25519 signature algorithm by design encourages address reuse.
+
+# Rationale and alternatives
+
+- The Ed25519 public-key signature system is a standard for modern signature schemes, as it offers high-speed high-security signatures.
+- This RFC aims for a seamless integration of Ed25519 signatures in the existing protocol. As such, the transaction structure as well as the resulting address format remain unchanged. While this offers the least possible development overhead to adapt existing W-OTS-only implementations, it also comes with a few drawbacks:
+  - W-OTS and Ed25519 addresses look exactly the same. This makes it impossible for the user or the system to apply any special considerations for the different signature schemes based on the addresses alone.
```

Comment:
Will Ed25519 addresses also have checksums when shown to the user? If so, systems could figure out whether the address is W-OTS or Ed25519 by recalculating the checksum (assuming Ed25519 address checksums are calculated differently from W-OTS address checksums)

Reply:
It was the goal of this RFC to not change anything in the current IOTA protocol transaction layout. As such, even Ed25519 addresses would need to be represented in 243 trits.
Unfortunately, due to the possibility of replay-attacks this entire idea is not feasible and Ed25519 needs to be combined with transaction layout changes, where we can indeed have different address versions